api/api/accounts/tests.py

```diff
@@ -0,0 +1,132 @@
+import pytest
+from django.contrib.auth import get_user_model
+from rest_framework.test import APITestCase
+from rest_framework import status
+
+User = get_user_model()
+userCreationEndpoint = '/users/'
+primaryTestUserMail = 'test@user.com'
+
+
+class UsersListViewTest(APITestCase):
+
+    @pytest.mark.django_db
+    def testUsersListView_testBaseUserModel_validEmailProvided_userSuccessfullyCreated(self):
+        # given
+        user = User.objects.create(
+            email=primaryTestUserMail, first_name='Test', last_name='User')
+        self.client.force_authenticate(user)
+        testUserEmail = 'test1@user.com'
+
+        # when
+        response = self.client.post(userCreationEndpoint, {'email': testUserEmail}, format='json')
+
+        # then
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(response.data['email'], testUserEmail)
+
+    @pytest.mark.django_db
+    def testUsersListView_invalidEmailsProvided_returnsHttp400(self):
+        # given
+        user = User.objects.create(
+            email=primaryTestUserMail, first_name='Test', last_name='User')
+        self.client.force_authenticate(user)
+
+        invalidTestMail1 = primaryTestUserMail
+        invalidTestMail2 = ''
+        invalidTestMail3 = 'testusercom'
```

Comment:
You can use [@pytest.mark.parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) to have test data defined separately from the test implementation.

This way test can focus on the actual logic for testing.